tests/connections/test_ssh_connection.py

```diff
@@ -46,3 +48,111 @@ def check_ssh_connection_for_default_values(self, ssh_connection):
         self.assertEqual(ssh_connection.port, DEFAULT_SSH_PORT)
         self.assertEqual(ssh_connection.username, DEFAULT_SSH_USERNAME)
         self.assertEqual(ssh_connection.password, DEFAULT_SSH_PASSWORD)
+
+    def test_connect_error(self):
+        ssh_connection = SSHConnection()
+        with self.assertRaises(GvmError):
+            ssh_connection.connect()
+
+    def test_connect(self):
+        with patch('paramiko.SSHClient') as SSHClientMock:
+            client_mock = SSHClientMock.return_value
+            client_mock.exec_command.return_value = ['a', 'b', 'c']
+            ssh_connection = SSHConnection()
+
+            ssh_connection.connect()
+            self.assertEqual(ssh_connection._stdin, 'a')
+            self.assertEqual(ssh_connection._stdout, 'b')
+            self.assertEqual(ssh_connection._stderr, 'c')
+
+    def test_disconnect(self):
+        with patch('paramiko.SSHClient') as SSHClientMock:
+            client_mock = SSHClientMock.return_value
+            client_mock.exec_command.return_value = ['a', 'b', 'c']
+            ssh_connection = SSHConnection()
+
+            ssh_connection.connect()
+            self.assertEqual(ssh_connection._stdin, 'a')
+            self.assertEqual(ssh_connection._stdout, 'b')
+            self.assertEqual(ssh_connection._stderr, 'c')
+
+            ssh_connection.disconnect()
+            # make sure the attributes have been deleted
+            with self.assertRaises(AttributeError):
+                type(ssh_connection._stdin)
+            with self.assertRaises(AttributeError):
+                type(ssh_connection._stdout)
+            with self.assertRaises(AttributeError):
+                type(ssh_connection._stderr)
+            with self.assertRaises(AttributeError):
+                type(ssh_connection._socket)
+
+            with self.assertRaises(AttributeError):
+                # disconnect twice should not work ...
+                ssh_connection.disconnect()
+
+            ssh_connection._socket = None
+            ssh_connection.disconnect()
+
+    def test_disconnect_os_error(self):
+        with patch('paramiko.SSHClient') as SSHClientMock:
+            client_mock = SSHClientMock.return_value
+            client_mock.exec_command.return_value = ['a', 'b', 'c']
+            client_mock.close.side_effect = OSError
+
+            ssh_connection = SSHConnection()
+            ssh_connection.connect()
+
+            with self.assertRaises(OSError):
+                ssh_connection.disconnect()
+
+    def test_send(self):
+        with patch('paramiko.SSHClient') as SSHClientMock:
+            client_mock = SSHClientMock.return_value
+            stdin = Mock()
+            stdin.channel.send.return_value = 4
+            client_mock.exec_command.return_value = [stdin, None, None]
+            ssh_connection = SSHConnection()
+
+            ssh_connection.connect()
+            ssh_connection.send("blah")
```

Comment:
I am missing some testing of the outcome here. It's just running code but not ensuring some behavior.